pkg/controllers/customeradmin/group_reconcile.go

```diff
@@ -0,0 +1,66 @@
+package customeradmin
+
+import (
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	protectedUsers = map[string]struct{}{
+		"customer-cluster-admin":  {},
+		"customer-cluster-reader": {},
+	}
+)
+
+func reconcileOneGroup(log *logrus.Entry, gc *graphrbac.GroupsClient, cli *openshiftClient, group graphrbac.ADGroup, kgroupName string, msGroupMembers []graphrbac.AADObject) error {
+	kubeGroup, err := cli.UserV1.Groups().Get(kgroupName, meta_v1.GetOptions{})
```

Comment:
We should be reading these from a local cache instead of going directly to the api server (similar to how the core controllers are working).

Reply:
Maybe OK for now, could be a follow-up.

Reply:
it can be a follow-up.  It's probably not wildly difficult to achieve: the controller-time framework already gives you a cached client - see the `client client.Client` member of `reconcileNamespace`